example/selection.js

```diff
@@ -12,29 +12,239 @@ import {
 
 const params = {
 
+	/** Selection tool: lasso (freehand shape) or box. */
 	toolMode: 'lasso',
+	/**
+	 * How triangles are marked for selection:
+	 * - intersection: if any part of the triangle is within the selection shape.
+	 * - centroid: if the center of the triangle is within the selection shape.
+	 * - centroid-visible: if the center of the triangle is within the selection shape and the triangle is visible.
+	 */
 	selectionMode: 'intersection',
+	/** Show selected triangles during the drag or only after the selection is completed. */
 	liveUpdate: false,
+	/** Select the whole mesh if one of its triangles was intersected. */
 	selectModel: false,
+	/** How to render the object: with solid material or edges only (wireframe). */
 	wireframe: false,
+	/** Use the BVH to speed up computations. */
 	useBoundsTree: true,
 
+	/** Show the boxes of the BVH. */
 	displayHelper: false,
+	/** The depth of BVH boxes to display, needs `displayHelper: true`. */
 	helperDepth: 10,
+	/** Continuously rotate the object. */
 	rotate: true,
 
 };
 
+/** Abstract class representing a selection using a pointer. */
+class Selection {
+
+	constructor() {
+
+		this.dragging = false;
+
+	}
+
+	handlePointerDown() {
+
+		this.dragging = true;
+
+	}
+	handlePointerUp() {
+
+		this.dragging = false;
+
+	}
+	handlePointerMove() {}
+
+	get points() {
+
+		return [];
+
+	}
+
+	/** Convert absolute screen coordinates `x` and `y` to relative coordinates in range [-1; 1]. */
+	static normalizePoint( x, y ) {
+
+		return [
+			( x / window.innerWidth ) * 2 - 1,
+			- ( ( y / window.innerHeight ) * 2 - 1 ),
+		];
+
+	}
+
+}
+
+
+const tempVec0 = new THREE.Vector2();
+const tempVec1 = new THREE.Vector2();
+const tempVec2 = new THREE.Vector2();
+/** Selection that adds points on drag and connects the start and end points with a straight line. */
+class LassoSelection extends Selection {
```

Comment:
I think we can also include generation of the line geometry for the screen shape in this function, as well (see "selectionShape").